### ISSUE TITLE
linter names are hard-coded

### ISSUE THREAD

**Issue by AshesITR** [jimhester/lintr#664]
Actual:
```r
as.data.frame(lintr::lint("A = 42\n", linters = list(lala = lintr::assignment_linter)))
>  filename line_number column_number  type                        message   line
>1   <text>           1             3 style Use <-, not =, for assignment. A = 42
>             linter
>1 assignment_linter
```

Expected:
```r
as.data.frame(lintr::lint("A = 42\n", linters = list(lala = lintr::assignment_linter)))
>  filename line_number column_number  type                        message   line
>1   <text>           1             3 style Use <-, not =, for assignment. A = 42
>             linter
>1 lala
```

**Comment by AshesITR** [comment]
This feature is particularly useful in combination with #660 because it allows for a straight-forward way to specify exclusions even for parameterised linters with multiple instances running, e.g. multiple `unwanted_function_linters` with different blacklists.
